### ISSUE TITLE
with postgres.Backend job.RunAfter is not always honored

### ISSUE THREAD

**Issue by pconstantinou** [acaloiaro/neoq#120]
Steps to reproduce:
- Create a queue and handler
- Add a job that has `RunAfter: time.Now().Add(time.Hour)`

Expected:
- Job doesn't run for an hour

Actual:
- Job runs immediately

In a fork, I added logging to confirm:
https://github.com/pconstantinou/neoq/blob/main/backends/postgres/postgres_backend.go#L928

**Comment by acaloiaro** [comment]
Are you able to reproduce with a shorter time frame? E.g. 1 minute?

**Comment by pconstantinou** [comment]
Yes. I could give you a test if you’d like.

**Comment by acaloiaro** [comment]
@pconstantinou I think we're going to need that reproduction test. I just added [TestFutureJobProcessing](https://github.com/acaloiaro/neoq/blob/main/backends/postgres/postgres_backend_test.go#L779) to cover this case, and it's currently passing. In my manual testing, I tested a few jobs with `RunAfter` ranging between 30 seconds and 10 minutes, and all jobs ran promptly at the specified `RunAfter`.  

There must be something unique about your situation that this test doesn't cover. 

The only thing I can think of is that you set the `JobCheckInterval` on `neoq.Config` to a value greater than the time between `time.Now()` and `RunAfter`. It's default is `1 second`. I've considered not exposing `JobCheckInterval` for this reason. If this does turn out to be the case, it's probably best addressed with documentation. Or perhaps not exposing `JobCheckInterval` at all.
